core/extraction/function_extractor.py

```
@@ -217,6 +166,65 @@
+    """Handles extraction of functions from Python source code."""
+
+    def __init__(
+        self,
+        context,
+        correlation_id: Optional[str] = None,
+    ) -> None:
+        """Initialize the function extractor."""
+        super().__init__(context, correlation_id)
+
+    async def extract_functions(
+        self, nodes: Union[ast.AST, List[ast.AST]], module_metrics: Any
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)